facebook-commerce.php

```diff
@@ -705,10 +722,13 @@ public function allow_full_batch_api_sync() {
 		 * @return boolean True if full batch sync is safe.
 		 * @since 2.6.1
 		 */
-		return apply_filters(
+		return apply_filters_deprecated(
 			'facebook_for_woocommerce_allow_full_batch_api_sync',
-			$default_allow_sync,
-			$this->get_product_count()
+			array(
+				$default_allow_sync,
+				$this->get_product_count(),
```

Comment:
If we've set the filter `facebook_for_woocommerce_block_full_batch_api_sync` to return true this code won't be reached. But in normal situations without any filters, `apply_filters_deprecated` would return early (without reporting a deprecated notice), but we'd still get the performance hit of counting the products. Would it be beneficial to add an additional `has_filter` call to prevent that?

Reply:
That is a good idea - implemented.